challenges/day14_script.js

```diff
@@ -0,0 +1,115 @@
+const fs = require('fs');
+
+const fileName = 'day14_input.txt';
+
+class Command {
+	constructor(memAddr, value) {
+		this.memoryAddress = this.createBinary(memAddr);
+		this.value = value;
+	}
+
+	createBinary(value) {
+		let x = value.toString(2);
+
+		return x.padStart(36, '0');
+	}
+}
+
+function run() {
+	const memory = populateMemory();
+	const sumMemoryValues = addValues(memory);
+
+	console.log(sumMemoryValues);
+}
+
+function populateMemory() {
+	const rl = fs.readFileSync(fileName).toString("utf-8").split("\n");
+	let memory = new Array();
+
+	let currentMask = "";
+	for (var i = 0; i < rl.length; i++) {
+		const line = rl[i].trim();
+
+		if (isMask(line)) {
+			currentMask = parseMask(line);
+		}
+		else {
+			let command = parseCommand(line);
+			let addresses = applyMask(command.memoryAddress, currentMask);
+
+			addresses.forEach(addr => memory[parseInt(addr, 2)] = command.value);
```

Comment:
This was the issue - array max index becomes 2^36. Iterators like filter, forEach only operate on the first 2^32 indices, so there were values not being added to the final sum.